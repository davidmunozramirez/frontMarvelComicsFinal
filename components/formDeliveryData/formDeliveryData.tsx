import { yupResolver } from "@hookform/resolvers/yup";
import { Box, Grid, Stack, TextField } from "@mui/material";
import { FormProvider, useForm, useFormContext } from "react-hook-form";
import DeliverytSchema from "dh-marvel/features/checkout/schemas/deliverySchema";
import { FC, useEffect } from "react";
import ControlledInput from "../controlledInput/controlledInput";
import { DeliveryDataType } from "dh-marvel/features/checkout/deliveryData.types";
import useOrder from "dh-marvel/features/formContext/useOrder";
import StepperNavigation from "../stepperNavigator/stepperNavigator";

export type RegisterFormProps = {
  activeStep: number;
  handleNext: () => void;
  onPrevClick: () => void;
};
export const FormDeliveryData: FC<RegisterFormProps> = ({
  activeStep,
  handleNext,
  onPrevClick,
}: RegisterFormProps) => {
  const { dispatch } = useOrder();

  const methods = useForm<DeliveryDataType>({
    resolver: yupResolver(DeliverytSchema),
    defaultValues: {
      address1: "Test main address",
      address2: "Test address complement",
      city: "Test City",
      state: "Test State",
      zipCode: "1234",
    },
  });
  const { watch, setFocus, handleSubmit, register } = methods;
  const address1 = watch("address1");
  const address2 = watch("address2");
  const city = watch("city");
  const state = watch("state");
  const zipCode = watch("zipCode");

  const onSubmit = (data: DeliveryDataType) => {
    dispatch({
      type: "SET_ADDRESS",
      payload: data,
    });
    handleNext();
  };

  const handleonPrevClick = () => {
    onPrevClick();
  };

  useEffect(() => {
    setFocus("address1");
  }, []);


  return (
    <Box sx={{ m: 2 }}>
      <Stack spacing={2}>
        <FormProvider {...methods}>
          <form onSubmit={handleSubmit(onSubmit)}>
            <ControlledInput
              name={"address1"}
              // {...register("address1", {required: true})}
              label={"Address 1"} />
            <ControlledInput
              name={"address2"}
              // {...register("address2", {required: false})}
              label={"Address 2"} />
            <ControlledInput
              name={"city"}
              // {...register("city", {required: true})}
              label={"City"} />
            <Grid container spacing={2}>
              <Grid item>
                <ControlledInput
                  name={"state"}
                  // {...register("state", {required: true})}
                  label={"State"} />
              </Grid>
              <Grid item>
                <ControlledInput
                  name={"zipCode"}
                  // {...register("zipCode", {required: true})}
                  // type={"number"}
                  label={"Zip Code"} />
              </Grid>
            </Grid>
          </form>
        </FormProvider>
        <StepperNavigation
          activeStep={activeStep}
          onPrevClick={handleSubmit(handleonPrevClick)}
          onNextClick={handleSubmit(onSubmit)}
        />
        <div>
          <h1>Validate your delivery data</h1>
          <h4>Address 1: {address1}</h4>
          <h4>Address 2: {address2}</h4>
          <h4>City: {city}</h4>
          <h4>State: {state}</h4>
          <h4>Zip Code: {zipCode}</h4>
        </div>
      </Stack>
    </Box>
  );
};
