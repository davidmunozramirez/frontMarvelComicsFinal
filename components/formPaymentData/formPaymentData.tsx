import { yupResolver } from "@hookform/resolvers/yup";
import { Alert, Box, Grid, Snackbar, Stack, TextField } from "@mui/material";
import { FormProvider, useForm, useFormContext } from "react-hook-form";
import PaymentSchema from "dh-marvel/features/checkout/schemas/paymentSchema";
import { FC, useEffect, useState } from "react";
import useOrder from "dh-marvel/features/formContext/useOrder";
import { PaymentDataType } from "dh-marvel/features/checkout/paymentData.type";
import ControlledInput from "../controlledInput/controlledInput";
import StepperNavigation from "../stepperNavigator/stepperNavigator";
import { postForm } from "dh-marvel/features/checkout/postForm";
import { useRouter } from "next/router";

export type RegisterFormProps = {
  activeStep: number;
  handleNext: () => void;
  onPrevClick: () => void;
  idSnackbar: any;
};
export const FormPaymentData: FC<RegisterFormProps> = ({
  activeStep,
  handleNext,
  onPrevClick,
  idSnackbar,
}: RegisterFormProps) => {
  const { dispatch, state } = useOrder();
  const router = useRouter();
  //Snackbar
  const [openSnackbar, setOpenSnackbar] = useState<any>(false);
  const [messageSnackbar, setMessageSnackbar] = useState<any>("");
  const [infoSnackbar, setInfoSnackbar] = useState<any>();
  //
  const methods = useForm<PaymentDataType>({
    resolver: yupResolver(PaymentSchema),
    defaultValues: {
      number: "1234567890123456",
      cvc: "123",
      expDate: "01/31",
      nameOnCard: "TEST CREDIT CARD",
    },
  });
  const { watch, setFocus, handleSubmit, register, formState: { errors } } = methods;
  const number = watch("number");
  const cvc = watch("cvc");
  const expDate = watch("expDate");
  const nameOnCard = watch("nameOnCard");

  const onSubmit = (data: PaymentDataType) => {
    dispatch({
      type: "SET_CARD",
      payload: data,
    });
    postForm({ ...state.order, card: data }).then((data) =>
      setInfoSnackbar(data)
    );
    handleNext();
  };

  useEffect(() => {
    console.log("dentro del use effect", infoSnackbar);

    if (infoSnackbar?.error) {
      console.log("bsatman", infoSnackbar);

      setMessageSnackbar(infoSnackbar?.message);
      setOpenSnackbar(true);
    }
    if (infoSnackbar?.data) {
      console.log("dentro del if del use effect", infoSnackbar);

      const handledData = { inital: infoSnackbar.data, data: idSnackbar };
      router.push(
        {
          pathname: `/confirmation/${idSnackbar}`,
          query: { data: JSON.stringify(handledData) },
        });
    }
  }, [infoSnackbar]);
  console.log("fuera del use effect", infoSnackbar);

  const handleonPrevClick = () => {
    onPrevClick();
  };

  useEffect(() => {
    setFocus("number");
  }, []);

  const handleClose = (
    event?: React.SyntheticEvent | Event,
    reason?: string
  ) => {
      if (reason === "clickaway") {
        return;
      }
    setOpenSnackbar(!false);
  };

  return (
    <Box sx={{ m: 2 }}>
      <Stack spacing={2}>
        <Snackbar
          open={openSnackbar}
          autoHideDuration={6000}
          onClose={handleClose}
        >
          <Alert onClose={handleClose} severity="error" sx={{ width: "100%" }}>
            {messageSnackbar}
          </Alert>
        </Snackbar>

        <FormProvider {...methods}>
          <form onSubmit={handleSubmit(onSubmit)}>
            <ControlledInput
              // name={"nameOnCard"}
              {...register("nameOnCard", {required: true} )}
              label={"Name On Card"}
              inputProps={{"data-testid": "nameOnCard"}} />
            {errors["nameOnCard"]?.type === "required" && <p>This field is required</p> }
            <ControlledInput
              // name={"number"}
              {...register("number", {required: true} )}
              label={"Credit Card"}
              type={"password"}
              inputProps={{"data-testid": "password"}} />
            <Grid container spacing={2}>
              <Grid item>
                <ControlledInput
                  // name={"expDate"}
                  {...register("expDate", {required: true} )}
                  label={"Expedition Date"}
                  inputProps={{"data-testid": "expDate"}}/>
              </Grid>
              <Grid item>
                <ControlledInput
                  // name={"cvc"}
                  {...register("cvc", {required: true} )}
                  label={"CVC"}
                  inputProps={{"data-testid": "cvc"}}/>
              </Grid>
            </Grid>
          </form>
        </FormProvider>
        <StepperNavigation
          activeStep={activeStep}
          onPrevClick={handleSubmit(handleonPrevClick)}
          onNextClick={handleSubmit(onSubmit)}
        />
        <div>
          <h1>Validate your payment data</h1>
          Name: {nameOnCard}
          <br />
          Number: {number.replace(number[0], "*").replace(number[1], "*").replace(number[2], "*").replace(number[3], "*").replace(number[4], "*").replace(number[5], "*").replace(number[6], "*").replace(number[7], "*").replace(number[8], "*").replace(number[9], "*").replace(number[10], "*").replace(number[11], "*")}
          <br />
          Expiration date: {expDate}
          <br />
          CVC: {cvc}
        </div>
      </Stack>
    </Box>
  );
};

{
  /* <TextField
{...register("ccname")}
required
fullWidth
id="ccname"
label="Name as it appears on your card"
name="ccname"
autoComplete="ccname"
helperText={errors?.ccname ? String(errors?.ccname?.message) : ""}
// onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
//   setInputsccname(event.target.value);
// }}
/> */
}
