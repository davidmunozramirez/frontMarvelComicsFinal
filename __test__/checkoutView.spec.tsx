import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  OrderProvider,
  OrderState,
} from "dh-marvel/features/formContext/OrderContext";
import useOrder from "dh-marvel/features/formContext/useOrder";
import { FormDeliveryData } from "dh-marvel/components/formDeliveryData/formDeliveryData";
import { FormPaymentData } from "dh-marvel/components/formPaymentData/formPaymentData";
import { StepperNavigationProps } from "dh-marvel/components/stepperNavigator/stepperNavigator";
import { FormPersonalData } from "../components/formPersonalData/formPersonalData";
import { CheckoutView } from '../components/checkoutView/checkoutView';

describe('CheckoutView', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<CheckoutView
        title={"Test Title"}
        image={"Test Image"}
        price={70}
        id={9876543210}
        idSnackbar={123456789}
    />);
    expect(asFragment()).toMatchSnapshot();
  })
})

describe("when rendering checkoutView with props", () => {
    it("should render some checkoutView texts", () => {
      const mockHandleNext = jest.fn();
      render(
        <>
          <CheckoutView
            title={"Test Title"}
            image={"Test Image"}
            price={70}
            id={9876543210}
            idSnackbar={123456789}
          />
        </>
      );
      const text1 = screen.queryByText(/Test Title/i);
      const text2 = screen.queryByText(/70/i);
      // const buttonBuyNow = screen.getByRole('button', {name: "Buy Now"})
      // const collapsibleDescription = screen.getByRole('button', {name: "Description"})
      // userEvent.click(screen.getByRole("button", { name: "Description" }));
      
      expect(text1).toBeInTheDocument();
      expect(text2).toBeInTheDocument();
      // expect(buttonBuyNow).toBeInTheDocument();
      // expect(collapsibleDescription).toBeInTheDocument();
    });

    it("should render checkoutView with formPersonalData", async () => {
        const mockHandleNext = jest.fn();
        const mockHandleBack = jest.fn();
        render(
          <>
            <CheckoutView
              title={"Test Title"}
              image={"Test Image"}
              price={70}
              id={9876543210}
              idSnackbar={123456789}
            >
                <FormPersonalData
                    activeStep={0}
                    handleNext={mockHandleNext}>
                </FormPersonalData>
            </CheckoutView>
          </>
        );
        
        userEvent.type(screen.getByRole("textbox", { name: "Name" }), "Test Name");
        userEvent.type(
            screen.getByRole("textbox", { name: "Last Name" }),
            "Test Last Name"
        );
        userEvent.type(
            screen.getByRole("textbox", { name: "Email" }),
            "testemail@email.com"
        );
        userEvent.click(screen.getByRole("button", { name: "Next" }));

        await waitFor(() => {
            expect(mockHandleNext).toBeCalled();
        });
        expect(mockDispatch).toBeCalledWith({
            payload: {
            name: "Test Name",
            lastname: "Test Last Name",
            email: "testemail@email.com",
            },
            type: "SET_CUSTOMER",
        });
    });

    it("should render checkoutView with formDeliveryData", async () => {
        const mockHandleNext = jest.fn();
        const mockHandleBack = jest.fn();
        render(
          <>
            <CheckoutView
              title={"Test Title"}
              image={"Test Image"}
              price={70}
              id={9876543210}
              idSnackbar={123456789}
            >
                <FormPersonalData
                    activeStep={0}
                    handleNext={mockHandleNext}>
                </FormPersonalData>
                
                <FormDeliveryData>
                    activeStep={0}
                    handleNext={mockHandleNext}
                    onPrevClick={mockHandleBack}    
                </FormDeliveryData>
            </CheckoutView>
          </>
        );
        
        userEvent.type(
            screen.getByRole("textbox", { name: "Address 1" }),
            "Test Main Address"
          );
          userEvent.type(
            screen.getByRole("textbox", { name: "Address 2" }),
            "Test address complement"
          );
          userEvent.type(
            screen.getByRole("textbox", { name: "City" }),
            "Test City"
          );
          userEvent.type(screen.getByRole("textbox", { name: "State" }), "Test State");
          userEvent.type(screen.getByRole("textbox", { name: "Zip Code" }), "1234");
    
          userEvent.click(screen.getByRole("button", { name: "Previous" }));
    
          await waitFor(() => {
            expect(mockHandleBack).toBeCalled();
          });
    });

    it("should render checkoutView with formPaymentData", async () => {
        const mockHandleNext = jest.fn();
        const mockHandleBack = jest.fn();
        render(
          <>
            <CheckoutView
              title={"Test Title"}
              image={"Test Image"}
              price={70}
              id={9876543210}
              idSnackbar={123456789}
            >
                <FormPaymentData>
                    activeStep={2}
                    handleNext={mockHandleNext}
                    onPrevClick={mockHandleBack}
                    idSnackbar={undefined}
                </FormPaymentData>
            </CheckoutView>
          </>
        );
        
        userEvent.type(
            screen.getByRole("textbox", { name: "Name On Card" }),
              "TEST CREDIT CARD"
          );
          userEvent.type(
            screen.getByTestId("password"),
            "1234567890123456"
          );
          userEvent.type(
            screen.getByRole("textbox", { name: "Expiration Date (MM/YY)" }),
            "01/31"
          );
          userEvent.type(screen.getByRole("textbox", { name: "CVC" }), "123");
    
          userEvent.click(screen.getByRole("button", { name: "Previous" }));
    
          await waitFor(() => {
            expect(mockHandleBack).toBeCalled();
          });
        //   expect(mockDispatch).toBeCalledWith({
        //     payload: {
        //       number: "1234567890123456",
        //       cvc: "123",
        //       expDate: "01/31",
        //       nameOnCard: "TEST CREDIT CARD",
        //     },
        //     type: "SET_CARD",
        //   });
    });
})


const mockStepperNavigationProps = jest.fn();
jest.mock("dh-marvel/components/stepperNavigator/stepperNavigator", () =>
  jest.fn((props: StepperNavigationProps) => {
    mockStepperNavigationProps(props);
    return (
      <div>
        StepperNavigation: {props.activeStep}
        <div>
          <button onClick={props.onPrevClick}>Previous</button>
          <button onClick={props.onNextClick}>Next</button>
        </div>
      </div>
    );
  })
);

jest.mock("dh-marvel/features/formContext/useOrder");
const mockUseOrder = useOrder as jest.MockedFunction<typeof useOrder>;
const mockDispatch = jest.fn();
mockUseOrder.mockReturnValue({
  state: {
    order: {
        address: {
          nameOnCard: "TEST CREDIT CARD",
          number: "1234567890123456",
          expDate: "01/31",
          cvc: "123",
        },
    },
  } as unknown as OrderState,
  dispatch: mockDispatch,
});

describe("should render checkoutView with formDelivery", () => {
  describe("when rendering default form", () => {
    it("should render the validation text", () => {
      const mockHandleNext = jest.fn();
      const mockHandleBack = jest.fn();
      render(
        <OrderProvider>
          <FormDeliveryData
            activeStep={0}
            handleNext={mockHandleNext}
            onPrevClick={mockHandleBack}
          />
        </OrderProvider>
      );
      const validator = screen.queryByText("Validate your delivery data");
      expect(validator).toBeInTheDocument();
    });

    it("should render button next", () => {
      const mockHandleNext = jest.fn();
      const mockHandleBack = jest.fn();
      render(
        <OrderProvider>
          <FormDeliveryData
            activeStep={0}
            handleNext={mockHandleNext}
            onPrevClick={mockHandleBack}
          />
        </OrderProvider>
      );
      const buttonN = screen.getByRole("button", { name: "Next" });
      expect(buttonN).toBeInTheDocument();
      const buttonB = screen.getByRole("button", { name: "Previous" });
      expect(buttonB).toBeInTheDocument();
    });
  });

  describe("when rendering submitting form", () => {
    it("should hit the dispatch", async () => {
      const mockHandleNext = jest.fn();
      const mockHandleBack = jest.fn();
      render(
        <OrderProvider>
          <FormDeliveryData
            activeStep={0}
            handleNext={mockHandleNext}
            onPrevClick={mockHandleBack}
          />
        </OrderProvider>
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "Address 1" }),
        "Test Main Address"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "Address 2" }),
        "Test address complement"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "City" }),
        "Test City"
      );
      userEvent.type(screen.getByRole("textbox", { name: "State" }), "Test State");
      userEvent.type(screen.getByRole("textbox", { name: "Zip Code" }), "1234");

      expect(screen.queryByText("Address 1: Test main address")).toBeInTheDocument();
      expect(screen.queryByText(/Address 2: Test address complement/i)).toBeInTheDocument();

      userEvent.click(screen.getByRole("button", { name: "Next" }));

      await waitFor(() => {
        expect(mockHandleNext).toBeCalled();
      });
      expect(mockDispatch).toBeCalledWith({
        payload: {
          address1: "Test main address",
          address2: "Test address complement",
          city: "Test City",
          state: "Test State",
          zipCode: "1234",
        },
        type: "SET_ADDRESS",
      });
    });

    it("click on previous", async () => {
      const mockHandleNext = jest.fn();
      const mockHandleBack = jest.fn();
      render(
        <OrderProvider>
          <FormDeliveryData
            activeStep={0}
            handleNext={mockHandleNext}
            onPrevClick={mockHandleBack}
          />
        </OrderProvider>
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "Address 1" }),
        "Test Main Address"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "Address 2" }),
        "Test address complement"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "City" }),
        "Test City"
      );
      userEvent.type(screen.getByRole("textbox", { name: "State" }), "Test State");
      userEvent.type(screen.getByRole("textbox", { name: "Zip Code" }), "1234");

      // expect(screen.queryByText("Address 1: Test main address")).toBeInTheDocument();
      // expect(screen.queryByText(/Address 2: Test address complement/i)).toBeInTheDocument();

      userEvent.click(screen.getByRole("button", { name: "Previous" }));

      await waitFor(() => {
        expect(mockHandleBack).toBeCalled();
      });
     
    });

    it("click on previous", async () => {
      const mockHandleNext = jest.fn();
      const mockHandleBack = jest.fn();
      render(
        <OrderProvider>
          <FormDeliveryData
            activeStep={0}
            handleNext={mockHandleNext}
            onPrevClick={mockHandleBack}
          />
        </OrderProvider>
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "Address 1" }),
        "Test Main Address"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "Address 2" }),
        "Test address complement"
      );
      userEvent.type(
        screen.getByRole("textbox", { name: "City" }),
        "Test City"
      );
      userEvent.type(screen.getByRole("textbox", { name: "State" }), "Test State");
      userEvent.type(screen.getByRole("textbox", { name: "Zip Code" }), "1234");

      // userEvent.click(screen.getByRole("button", { name: "Previous" }));

      await waitFor(() => {
        expect(screen.queryByText("Address 1: Test main address")).toBeInTheDocument();
        expect(screen.queryByText(/Address 2: Test address complement/i)).toBeInTheDocument();
        expect(screen.queryByText("City: Test City")).toBeInTheDocument();
        expect(screen.queryByText(/State: Test State/i)).toBeInTheDocument();
        expect(screen.queryByText("Zip Code: 1234")).toBeInTheDocument();
      });
     
    });


  });
});


