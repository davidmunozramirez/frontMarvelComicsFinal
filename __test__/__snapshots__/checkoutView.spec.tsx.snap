// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutView should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      class="MuiStack-root css-fdaco8-MuiStack-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-14leyog-MuiTypography-root"
      >
        Checkout: Test Title
      </h2>
      <div
        class="MuiBox-root css-1ykdma4"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiStepper-root MuiStepper-horizontal css-m5vj9m-MuiStepper-root"
          >
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-active css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      1
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active css-1hv8oq8-MuiStepLabel-label"
                  >
                    Personal Data
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      2
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1hv8oq8-MuiStepLabel-label"
                  >
                    Delivery Adress
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      3
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1hv8oq8-MuiStepLabel-label"
                  >
                    Payment Infomation
                  </span>
                </span>
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root css-1c5ij41"
          >
            <div
              class="MuiStack-root css-1p5q5e5-MuiStack-root"
            >
              <form>
                <div
                  class="MuiBox-root css-1qm1lh"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for=":r0:"
                      id=":r0:-label"
                    >
                      Name
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r0:"
                        name="name"
                        type="text"
                        value="Test"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Name
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-1qm1lh"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for=":r1:"
                      id=":r1:-label"
                    >
                      Last Name
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r1:"
                        name="lastname"
                        type="text"
                        value="User"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Last Name
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-1qm1lh"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for=":r2:"
                      id=":r2:-label"
                    >
                      Email
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r2:"
                        name="email"
                        type="email"
                        value="test@user.com"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Email
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </form>
              <div>
                StepperNavigation: 0
                <div>
                  <button>
                    Previous
                  </button>
                  <button>
                    Next
                  </button>
                </div>
              </div>
              <div>
                <h1>
                  Validate your personal data
                </h1>
                Name: Test
                <br />
                Last Name: User
                <br />
                Email: test@user.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-j89rld-MuiPaper-root-MuiCard-root"
        >
          <img
            alt="Test Title"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1h2xq9u-MuiCardMedia-root"
            height="300"
            src="Test Image"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
            >
              Test Title
            </div>
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              70$
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
